### PR TITLE
Enhance the number parser to include support for parsing hexadecimal, binary, and octal literals.

### PR DESCRIPTION
**Pull request by ckganesan**
The current parser lacks support for the following numeric representations:

1. Octal literals.
2. Binary literals.
3. Hexadecimal literals with a '0X' prefix.

### CODE REVIEW

File: parser/parser.go [hunk 1]
```diff
@@ -306,19 +306,8 @@ func (p *parser) parseSecondary() Node {
...
-			if number > math.MaxInt {
-				p.error("integer literal is too large")
-				return nil
-			}
-			node := &IntegerNode{Value: int(number)}
-			node.SetLocation(token.Location)
-			return node
-		} else if strings.ContainsAny(value, ".eE") {
+		isHex := strings.HasPrefix(value, "0x") || strings.HasPrefix(value, "0X")
+		if !isHex && strings.ContainsAny(value, ".eE") {
```
Comment (antonmedv) [inline parser/parser.go:310]: We are checking here only for `!isHex`. What about others?

File: parser/parser_test.go [hunk 2]
```diff
@@ -36,6 +37,26 @@ func TestParse(t *testing.T) {
...
+			"0O45",
+			&IntegerNode{Value: 37},
+		},
+		{
+			"0b10",
+			&IntegerNode{Value: 2},
+		},
+		{
+			"0B101011",
+			&IntegerNode{Value: 43},
```
Comment (antonmedv) [inline parser/parser_test.go:58]: Let's also add test for invalid literals.

**Review by antonmedv — CHANGES_REQUESTED** [review]

File: parser/parser.go (hunk 1 above)
Comment (ckganesan) [reply parser/parser.go]: We are specifically excluding hexadecimal literals containing the letter 'E' when performing float conversions.

**Comment by ckganesan** [comment]
Hello @antonmedv, I kindly request your review of this pull request when you have a moment. Your insights would be greatly appreciated. Thank you in advance for your time and assistance!

**Review by antonmedv — APPROVED** [review]

**Pull request merged**